core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala

```diff
@@ -50,12 +52,22 @@ class HttpSamDAO(baseSamServiceURL: String, serviceAccountCreds: Credential)(imp
     }
   }
 
-  private def doSuccessOrFailureRequest(request: HttpRequest, userInfo: UserInfo) = {
+  private def doSuccessOrFailureRequest(request: HttpRequest, userInfo: UserInfo): RetryableFuture[Unit] = {
     retry(when401or500) { () =>
-      httpClientUtils.executeRequest(http, httpClientUtils.addHeader(request, authHeader(userInfo))).map { response =>
+      httpClientUtils.executeRequest(http, httpClientUtils.addHeader(request, authHeader(userInfo))).flatMap { response =>
         response.status match {
-          case s if s.isSuccess => ()
-          case f => throw new RawlsExceptionWithErrorReport(ErrorReport(f, s"Sam call to ${request.method} ${request.uri.path} failed with error ${response.entity}"))
+          case s if s.isSuccess => Future(())
+          case f =>
+            // attempt to propagate an ErrorReport from Sam. If we can't understand Sam's response as an ErrorReport,
+            // create our own error message.
+            import WorkspaceJsonSupport.ErrorReportFormat
+            toFutureTry(Unmarshal(response.entity).to[ErrorReport]) map {
+              case Success(err) =>
+                logger.error(s"Sam call to ${request.method} ${request.uri.path} failed with error $err")
+                throw new RawlsExceptionWithErrorReport(err)
+              case Failure(_) =>
+                throw new RawlsExceptionWithErrorReport(ErrorReport(f, s"Sam call to ${request.method} ${request.uri.path} failed with error ${response.entity}"))
+            }
```

Comment:
the extra `logger.error` in the `Success` case exists so that we are always logging the "Sam call ... failed" text. Since we propagate the raw `ErrorReport` in this case, if we kept just the `throw` we wouldn't log "Sam call...".